curator-recipes/src/main/java/org/apache/curator/framework/recipes/locks/InterProcessMultiLock.java

```diff
@@ -145,6 +145,9 @@ public boolean acquire(long time, TimeUnit unit) throws Exception
      * {@inheritDoc}
      *
      * <p>NOTE: locks are released in the reverse order that they were acquired.</p>
+     *	
+     * @throws Exception ZK errors, interruptions, current thread does not own the lock
```

Comment:
Seems right, but don't you need to add now same "@throws" text also for InterProcessMutex.release()? (This is also the actual code that throws IllegalMonitorStateException in case of different thread, and used internally here)

Reply:
@shayshim `InterProcessMutex.release()` already has that `@throws` in its javadoc. See https://github.com/emretetik96/curator/blob/77470cbbe0befa26500c6d18e6fc93c29f831725/curator-recipes/src/main/java/org/apache/curator/framework/recipes/locks/InterProcessMutex.java#L126

Reply:
@emretetik96 right, my mistake